app/routes/_layout+/assets._index.tsx

```diff
@@ -159,6 +170,27 @@ export async function loader({ context, request }: LoaderFunctionArgs) {
       plural: "assets",
     };
 
+    const headers = new Headers();
```

Comment:
I think everything from this line to line 192 should be abstracted into a function that returns the headers.